[linette-g] iP

linette-g opened the pull request:
## Rob
> “The way to get started is to quit talking and begin doing.” – Walt Disney

Rob is here to streamline your task management and make your life easier!

- **User-Friendly**: Easy to navigate and interact with
- **Flexible Task Management**: Create, mark and find tasks with ease
- **Text-Based Interface**: Communicate with Rob with straightforward text commands

All you need to do is,

1. download it from [here](https://github.com/linette-g/ip/releases/tag/v0.1)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks (including deadlines and events)
- [ ]  Advanced search functions (coming soon)
- [ ]  Reminders (coming soon)

Example code for the `main` method::
```java
public static void main(String[] args) {
        new Rob(FILE_PATH).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

linette-g committed:
Add skeletal version of chatbot Rob

linette-g committed:
Add echo

linette-g committed:
Add list

linette-g committed:
Add "Mark as Done" feature

linette-g committed:
Add ToDo, Event, Deadline

linette-g committed:
Add tests using the I/O redirection technique

linette-g committed:
Add exceptions

linette-g committed:
Add delete

linette-g committed:
Add automatic task saving and loading

linette-g committed:
Merge branch 'branch-Level-7'

linette-g committed:
Add date and time formatting

linette-g committed:
Merge branch 'branch-Level-8'

linette-g committed:
Add Ui, Storage, Parser and TaskList classes

linette-g committed:
Merge branch 'branch-A-MoreOOP'

linette-g committed:
Divide classes into packages

linette-g committed:
Merge branch 'branch-A-Packages'

linette-g committed:
Merge remote-tracking branch 'origin/add-gradle-support'

linette-g committed:
Add JUnit tests

linette-g committed:
Merge branch 'branch-A-JUnit'

linette-g committed:
Update build.gradle file

linette-g committed:
Add JavaDoc comments

linette-g committed:
Tweak the code to comply with a coding standard

linette-g committed:
Update more JavaDoc comments

linette-g committed:
Add method to find a task by searching for a keyword

linette-g committed:
Merge branch 'branch-A-JavaDoc'

linette-g committed:
Merge branch 'branch-A-CodingStandard'

linette-g committed:
Update parsing logic in Parser class

linette-g committed:
Merge branch 'branch-Level-9'

linette-g committed:
Fix Checkstyle issues

linette-g committed:
Merge branch 'branch-A-CheckStyle'

linette-g committed:
Add a GUI to the chatbot

Implemented a graphical user interface (GUI) for the chatbot
application using JavaFX. This update introduces a main window
with an input field for user queries and a display area for the
chatbot's responses.

linette-g committed:
Merge branch 'branch-Level-10'

muhdbhz commented on src/main/java/rob/Rob.java:
```diff
@@ -0,0 +1,151 @@
+                storage.saveTasks(tasks);
+                return delStr;
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else if (Objects.equals(command, "find")) {
+            try {
+                String keyword = parser.getFind();
+                List<Task> filteredList = tasks.searchTasks(keyword);
+                TaskList filteredTasks = new TaskList(filteredList);
+                return ui.showList(filteredTasks);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else {
+            try {
+                if (Objects.equals(command, "deadline")) {
+                    String desc = parser.getDesc();
+                    String day = parser.getDay();
+                    tasks.getTasks().add(new Deadline(desc, day));
+                    storage.saveTasks(tasks);
+                } else if (Objects.equals(command, "event")) {
+                    String desc = parser.getDesc();
+                    String from = parser.getFrom();
+                    String to = parser.getTo();
+                    tasks.getTasks().add(new Event(desc, from, to));
+                    storage.saveTasks(tasks);
+                } else if (Objects.equals(command, "todo")) {
+                    String desc = parser.getDesc();
+                    tasks.getTasks().add(new Todo(desc));
+                    storage.saveTasks(tasks);
+                } else {
+                    throw new DukeException("I'm sorry... I don't seem to understand.");
+                }
+                return ui.printText(tasks);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
```
Hi Linette! You could consider using a switch statement to reduce several lines of code here!

muhdbhz commented on src/main/java/rob/Ui.java:
```diff
@@ -0,0 +1,135 @@
+     */
+    public String delete(TaskList taskList, int taskNum) {
+        // echo
+        String echo = "____________________________________\n"
+                + "deleted: " + taskList.getTask(taskNum - 1) + "\n"
+                + "____________________________________\n";
```
I like that you break before the operator! It seems like many people have missed that as part of the coding standard. Nice! 👍

muhdbhz reviewed (commented):
LGTM! The code was easy to read and Linette has not made any coding standard violations. The review comments were just suggestions on how the code can be changed to improve readability even further. 👍 Good job!

SooYap commented on src/main/java/rob/Rob.java:
```diff
@@ -158,4 +62,90 @@ public int findTaskNum(String input) throws DukeException {
+     *
+     * @return The string containing the chatbot's response.
+     */
+    public String getResponse(String input) {
+        parser = new Parser(input);
+        parser.checkString();
+        String command = parser.getCommand();
+
+        if (Objects.equals(command, "bye")) {
+            return ui.getExit();
+        } else if (Objects.equals(command, "list")) {
+            return ui.showList(tasks);
+        } else if (Objects.equals(command, "mark")) {
+            try {
+                int taskNum = findTaskNum(input);
+                String markedStr = ui.mark(tasks, taskNum);
+                tasks.getTask(taskNum - 1).markAsDone();
+                storage.saveTasks(tasks);
+                return markedStr;
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else if (Objects.equals(command, "unmark")) {
+            try {
+                int taskNum = findTaskNum(input);
+                String unmarkedStr = ui.unmark(tasks, taskNum);
+                tasks.getTask(taskNum - 1).unmark();
+                storage.saveTasks(tasks);
+                return unmarkedStr;
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                return e.getMessage();
+            }
+        } else if (Objects.equals(command, "delete")) {
+            try {
+                int taskNum = findTaskNum(input);
```
Your variable names are appropriately named and adequately descriptive

SooYap commented on src/main/java/rob/Rob.java:
```diff
@@ -36,100 +34,6 @@ public Rob(String filePath) {
-     * Reads user input and processes commands such as:
-     * "bye", "list", "mark", "unmark" "delete", "deadline", "event", and "todo".
-     * Saves and loads tasks
-     */
```
JavaDoc comments are clear and allows me to understand the method

SooYap reviewed (commented):
Overall your code LGTM. It is understandable, easy to read, and methods and variables are well-named with appropriate JavaDoc comments.

linette-g committed:
Add assertions to document key assumptions

The code lacks assertions to document important assumptions.

Let's include checks to ensure critical conditions (e.g. a non-empty
file-path) are met.

linette-g committed:
Improve code quality by refactoring code

Refactoring improves code quality by making the codebase more
readable, efficient, and easier to maintain.

Let's use inline methods and refactor the existing `if-else`
statements to use a `switch` statement.

linette-g committed:
Merge pull request #2 from linette-g/branch-A-CodeQuality

Improve code quality by refactoring code

linette-g committed:
Merge remote-tracking branch 'origin/master' into branch-A-Assertions

linette-g committed:
Merge pull request #1 from linette-g/branch-A-Assertions

Add assertions to document key assumptions

linette-g committed:
Add the ability to recognize and deal with duplicate items

Let's,
- search the task list for any matching tasks
- prevent addition of new tasks using regular commands (e.g. todo …)
- force the addition using the force command (e.g. force todo …)

linette-g committed:
Merge branch 'branch-A-Extention'

linette-g committed:
Update file name

linette-g committed:
Update code quality

linette-g committed:
Add a User Guide

linette-g committed:
Merge branch 'branch-A-UserGuide'

linette-g committed:
Add product screenshot

linette-g committed:
Update main class configuration to `rob.Main`

linette-g committed:
Update to handle whitespace-only descriptions

linette-g committed:
Remove dead code

linette-g committed:
Update exception handling

linette-g committed:
Update product name

linette-g committed:
Add dependencies

linette-g committed:
Update build gradle
